dask/dataframe/io/sql.py

```diff
@@ -62,33 +72,37 @@ def read_sql_table(table, uri, npartitions, index_col, limits=None, columns=None
             minmax = pd.read_sql(q, engine)
             maxi, mini = minmax.iloc[0]
             if minmax.dtypes['max_1'].kind == "M":
-                npartitions = pd.date_range(
+                divisions = pd.date_range(
                     start=mini, end=maxi, freq='%iS' % (
                         (maxi - mini) / npartitions).total_seconds()).tolist()
+                divisions[-1] += datetime.timedelta(seconds=1)
             else:
-                npartitions = np.linspace(mini, maxi, npartitions + 1).tolist()
-                npartitions[-1] += 1
+                divisions = np.linspace(mini, maxi, npartitions + 1).tolist()
+                divisions[-1] += 1
         else:
             mini, maxi = limits
-            npartitions = np.linspace(mini, maxi, npartitions + 1).tolist()
-    columns = ([(table.columns[c] if isinstance(c, str) else c)
+            divisions = np.linspace(mini, maxi, npartitions + 1).tolist()
+    else:
+        divisions = npartitions
```

Comment:
This is the `else` to if the value is an integer. Perhaps it should be `elif isinstance(npartitions, (tuple, list, ndarray))` and raise an exception on else? 
Alternatively, could add a divisions= keyword, which overrides npartitions and limits, to remove the confusion.